### ISSUE TITLE
Web app improvements

### ISSUE THREAD

**Issue by georgefst** [cdepillabout/pretty-simple#81]
The [web](https://github.com/cdepillabout/pretty-simple/tree/web) branch contains [Miso](https://haskell-miso.org/) code for generating the app currently deployed at https://georgefst.github.io/pretty-simple. This is really just a proof of concept. Places to improve:

- [x] General prettiness - the current layout exposes the fact that I'm very much not a web developer.
  - EDIT: since #116, I don't think it's too bad (ironically, two years on from the above comment, frontend web development is a big part of my job).
- [ ] The initial settings for the buttons and sliders do not currently reflect the initial `OutputOptions`. I was unsure how to set the right HTML attributes, without them being constantly refreshed and causing flickering. Probably worth asking on the Miso Slack channel. See [this unused commit](https://github.com/cdepillabout/pretty-simple/tree/web-initial-value).
  - [ ] Relatedly, selecting an example from the drop-down menu should populate the text box.
- [ ] The settings should be reflected in the URL - I've found myself wanting to share a particular formatting of an expression.
- [x] It would be nice to have several example inputs to select from. Perhaps some of [these](https://hackage.haskell.org/package/pretty-simple/docs/Text-Pretty-Simple.html#g:8). At least one should show an effect from changing the `StringOutputStyle` dropdown (see #64).
- [ ] Sliders should display exact current value as text.
- [ ] Version bounds. I've been using GHC 9.2 with plain Cabal, and GHC 8.6 (!) with Nix, which may make it awkward to get this right. Also, because this isn't a library, and we mostly only care about the Nix build, which has everything pinned, this is much less of an issue than it would usually be.
- [x] Host at a more suitable URL. Probably https://cdepillabout.github.io/pretty-simple in order to match this repository.
- [x] Integrate with CI, so that it can be rebuilt and deployed whenever a release tag is pushed.
- [x] Link to the webpage prominently from Github and Hackage.
- [ ] Switch from GHCJS to GHC's own JS backend, or even WASM (see [examples](https://github.com/tweag/ghc-wasm-miso-examples)).

Note that, for development, everything can be built with just Cabal, which will use [jsaddle](https://hackage.haskell.org/package/jsaddle) to run a web server. Use `nix-build` to compile to JavaScript, using GHCJS.

Inspiration:
- https://jsoneditoronline.org/
- https://monadfix.com/shower/

**Comment by cdepillabout** [comment]
Thanks for putting this together.  It seems like there were a lot of good comments on reddit about the webview!

https://www.reddit.com/r/haskell/comments/iss9vo/ann_prettysimple_40/

**Comment by georgefst** [comment]
Further work on this (by me, anyway) is blocked on https://github.com/dmjio/miso/issues/639.

**Comment by cdepillabout** [comment]
@georgefst Wow, https://georgefst.github.io/pretty-simple/ is really nice!

I'd love if you sent a PR to get this all into the pretty-simple repo.  Or at least add this online page to the README of `pretty-simple`!

**Comment by georgefst** [comment]
> I'd love if you sent a PR to get this all into the pretty-simple repo.

https://github.com/cdepillabout/pretty-simple/pull/116

**Comment by georgefst** [comment]
This shouldn't have been closed. I think it's down to GitHub's parser seeing "close" on #116.

Anyway I've added another bullet point, about using GHC's WASM backend.
